pkg/util/kernel/fs.go

```diff
@@ -85,10 +87,13 @@ func IsDebugFSMounted() (bool, error) {
 		return false, fmt.Errorf("kprobe_events mount point(%s): wrong fs type(%s)", mi.Mountpoint, mi.FSType)
 	}
 
-	options := strings.Split(mi.Options, ",")
-	for _, option := range options {
-		if option == "ro" {
-			return false, fmt.Errorf("kprobe_events mount point(%s) is in read-only mode", mi.Mountpoint)
+	// on fargate, kprobe_events is mounted as ro
+	if !fargate.IsFargateInstance(context.Background()) {
```

Comment:
It's hard to say without seeing the rest of the code, but this feels like a weird level of code to check this. Maybe it would be more appropriate to return a bool to indicate if kprobe_events is `ro`

Reply:
I think this function is doing the equivalent of "is debugfs mounted correctly", so you have these additional checks other than debugfs being just mounted. So its just modifying the ro check for this environment.

Reply:
We do still need to check if debugfs is mounted; every other check in the function is valid for fargate. I can modify the function docs to say that the ro check is skipped on fargate.